fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/Ndv.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.doris.nereids.types.SmallIntType;
+import org.apache.doris.nereids.types.StringType;
+import org.apache.doris.nereids.types.TimeType;
+import org.apache.doris.nereids.types.TimeV2Type;
+import org.apache.doris.nereids.types.TinyIntType;
+import org.apache.doris.nereids.types.VarcharType;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * AggregateFunction 'ndv'. This class is generated by GenerateFunction.
+ */
+public class Ndv extends AggregateFunction
+        implements UnaryExpression, ExplicitlyCastableSignature, AlwaysNotNullable {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(BigIntType.INSTANCE).args(TinyIntType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(SmallIntType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(IntegerType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(BigIntType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(LargeIntType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(FloatType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DoubleType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DecimalV2Type.SYSTEM_DEFAULT),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DecimalV3Type.DEFAULT_DECIMAL32),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DecimalV3Type.DEFAULT_DECIMAL64),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DecimalV3Type.DEFAULT_DECIMAL128),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(BooleanType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(VarcharType.SYSTEM_DEFAULT),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(StringType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DateType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DateTimeType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DateV2Type.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(DateTimeV2Type.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(TimeType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(TimeV2Type.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(JsonType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(HllType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(BitmapType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(QuantileStateType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(NullType.INSTANCE),
+            FunctionSignature.ret(BigIntType.INSTANCE).args(CharType.INSTANCE)
```

Comment:
it is a bit hard to use here.
`CharType.INSTANCE` is return `CharacterType` actually.
We need use `CharType.SYSTEM_DEFAULT` just like `VarcharType`

Reply:
fixed.